specification/dartLangSpec.tex

```diff
@@ -2786,8 +2790,8 @@ \subsection{Setters}
 }
 
 \LMHash{}%
-It is a static warning if a setter declares a return type other than \VOID{}.
-It is a static warning if a class has
+It is a compile-time error if a setter declares a return type other than \VOID.
+It is a compile-time error if a class has
 a setter named \code{$v$=} with argument type $T$ and
 a getter named $v$ with return type $S$,
 and $S$ may not be assigned to $T$.
```

Comment:
So we do allow a `dynamic` getter and a non-top-type setter. Yey?

Reply:
Yes, this must be an oversight: The rule doesn't apply to non-class getters and setters. But the null-safety rules cover all setter/getter pairs, so we would presumably just leave legacy code as it is and get the cleanup with null-safety (rather than starting a breaking change process with legacy code).

Also the old rules allow a `dynamic` getter and any setter parameter type, but that's not a surprise: The nnbd feature spec explicitly says that it is _now_ a subtype rule, not an assignability rule.